tests/test_utilities.py

```diff
@@ -227,3 +229,159 @@ def test_get_course_jobs(self):
+        translate_text_mocked.return_value = None
+        expected_translated_description = utils.get_translated_course_description(COURSE_KEY, course_description)
+        translation_record = Translation.objects.filter(source_model_name='Course',
+                                                        source_model_field='full_description',
+                                                        source_record_identifier=COURSE_KEY).first()
+        assert translation_record is None
+        assert course_description == expected_translated_description
+        assert translate_text_mocked.call_count == 1
+
+    @mock.patch('taxonomy.utils.translate_text')
+    def test_get_translated_course_description_error_for_existing_record(self, translate_text_mocked):
+        """
+        Validate that `get_translated_course_description` returns actual course_description
+         if translate_text method returns None and does not update Translation object.
+        """
+        translate_text_mocked.return_value = None
+        course_description = "abc def"
+        new_course_description = "jhi qlm"
+        translated_course_description = "different text"
+        trans = factories.TranslationFactory(source_record_identifier=COURSE_KEY,
+                                             source_model_field='full_description',
+                                             source_model_name='Course',
+                                             source_text=course_description,
+                                             translated_text=translated_course_description,
+                                             translated_text_language=ENGLISH,
+                                             source_language=ENGLISH,
+                                             )
+
+        expected_translated_description = utils.get_translated_course_description(COURSE_KEY, new_course_description)
+        translation_record = Translation.objects.filter(source_model_name='Course',
+                                                        source_model_field='full_description',
+                                                        source_record_identifier=COURSE_KEY).first()
+
+        assert new_course_description == expected_translated_description
+        assert trans.source_text == translation_record.source_text
+        assert trans.translated_text == translation_record.translated_text
+        assert translate_text_mocked.call_count == 1
+
+    @mock.patch('taxonomy.utils.translate_text')
+    def test_get_translated_course_description_success_for_new_record(self, translate_text_mocked):
+        """
+        Validate that `get_translated_course_description` created Translation object if not already
+        exist with the translated course description.
+        """
+        course_description = "abc def"
+        translated_course_description = "different text"
+        translate_text_mocked.return_value = {'SourceLanguageCode': ENGLISH,
```

Comment:
A Scenario where SourceLanguageCode is other than ENGLISH is remaining, That is why you coverage is wailing

Reply:
One case was added earlier for updating the course description record. A new test case is also added which will be for the newly created records.